File: test/corpus/basic/003-word-boundaries/case.go
package main

import (
	"fmt"
	"github.com/rivo/uniseg"
)

func main() {
	// Test cases for word boundary detection
	testCases := []string{
		"Hello, world!",
		"Test 123. More text!",
		"こんにちは世界",       // Japanese
		"Hello\nWorld",
		"word-with-hyphens",
		"CamelCaseWord",
		"user@example.com",
		"U.S.A.",
		"",
		"   ",
		"a",
		"café naïve résumé",   // Accented characters
	}

	for _, test := range testCases {
		state := -1
		var word string
		remaining := test
		result := []string{}
		
		for len(remaining) > 0 {
			word, remaining, state = uniseg.FirstWordInString(remaining, state)
			if word == "" {
				break
			}
			result = append(result, word)
		}
		
		fmt.Printf("%d\n", len(result))
		for _, w := range result {
			if w == " " {
				fmt.Print("(SPACE)")
			} else if w == "\n" {
				fmt.Print("(LF)")
			} else if w == "\t" {
				fmt.Print("(TAB)")
			} else {
				fmt.Printf("(%s)", w)
			}
		}
		fmt.Println()
	}
}

File: test/corpus/basic/003-word-boundaries/case.ts
import { firstWordInString } from '#src/index.js';

function main() {
  // Test cases for word boundary detection
  const testCases = [
    'Hello, world!',
    'Test 123. More text!',
    'こんにちは世界', // Japanese
    'Hello\nWorld',
    'word-with-hyphens',
    'CamelCaseWord',
    'user@example.com',
    'U.S.A.',
    '',
    '   ',
    'a',
    'café naïve résumé', // Accented characters
  ];

  for (const test of testCases) {
    let state = -1;
    let remaining = test;
    const result: string[] = [];

    while (remaining.length > 0) {
      const wordResult = firstWordInString(remaining, state);
      if (wordResult.segment === '') {
        break;
      }
      result.push(wordResult.segment);
      remaining = wordResult.remainder;
      state = wordResult.newState;
    }

    process.stdout.write(`${result.length}\n`);
    for (const w of result) {
      if (w === ' ') {
        process.stdout.write('(SPACE)');
      } else if (w === '\n') {
        process.stdout.write('(LF)');
      } else if (w === '\t') {
        process.stdout.write('(TAB)');
      } else {
        process.stdout.write(`(${w})`);
      }
    }
    process.stdout.write('\n');
  }
}

main();
